app/models/sqs_message.rb

```diff
@@ -0,0 +1,48 @@
+class SqsMessage
+  def initialize(thesis)
+    @thesis = thesis
+    @package_id = "etd_#{@thesis.id}"
+    @metadata_uri = thesis.dspace_metadata.blob.url
+  end
+
+  def message_attributes
+    attributes = {}
+    attributes['PackageID'] = { 'DataType' => 'String', 'StringValue' => @package_id }
+    attributes['SubmissionSource'] = { 'DataType' => 'String', 'StringValue' => 'ETD' }
+    attributes['OutputQueue'] = { 'DataType' => 'String', 'StringValue' => ENV['OUTPUT_QUEUE_NAME'].to_s }
+    attributes
+  end
+
+  def message_body
+    body = {}
+    body['SubmissionSystem'] = 'DSpace@MIT'
+    body['CollectionHandle'] = collection_handle
+    body['MetadataLocation'] = @metadata_uri
+    body['Files'] = map_files
+
+    # SQS requires the MessageBody to be a string
+    body.to_json
+  end
+
+  def map_files
+    @thesis.files.map do |f|
+      {
+        'BitstreamName' => f.blob.filename.to_s, # is this also the filename in DSpace?
+        'FileLocation' => f.blob.url,
+        'BitstreamDescription' => -> { f.description ? [f.purpose, f.description].join(' ') : f.purpose }.call
```

Comment:
Note to reviewer: using a lambda here seemed like the most sensible way to deal with a Rubocop style issue. The alternative was string interpolation, which Rubocop generally doesn't like in this sort of situation (and I tend to agree). I'm happy to revert to interpolation if we think this is significantly less readable.

Reply:
I don't see rubocop complaining about this syntax that I think does the same thing?

```
'BitstreamDescription' => "#{f.purpose} #{f.description}".strip
```

Reply:
Also, are we sure we want to put the raw purpose in here? `thesis_pdf some description` seems weird to me (we can always adjust later if we weren't given a lookup table for what to actually put in here)

Reply:
Ha! Yeah, you're right. Now I want to know why Rubocop was flagging that approach for me last week...must've been a typo.

As for the purpose and description, my reading of the spec is that the bitstream description contains at minimum the file purpose, with the description concatenated if present. I got that from lines 17-18 in the mapping spreadsheet: https://docs.google.com/spreadsheets/d/1JhNvD_0VXCacQRtRgWN3RUX6-OOeTJKv0AbnZFzjgHw/edit#gid=17335869. I think there are still crosswalk issues to resolve with those fields, though, so that may change later.

Reply:
Yeah I think what we have here is fine, but the literal wording "thesis_pdf" seems suspiciously not super user friendly and we'll probably want someone to map us to preferred wording eventually. Maybe a ticket to confirm wording on the options here would be good or we can just ask stakeholders to confirm it when we send samples into test dspace later on.

Reply:
Ah, now I understand what you're saying. So maybe a convenience method that translates the raw purposes into a more readable format?